### PR TITLE
RDI-2141 Fix capturing performance problem in the webcanvassrc element

### PR DESCRIPTION
**Pull request by aslobodeniuk**

### CODE REVIEW

File: gst.wasm/subprojects/samples/webcanvassrc/webcanvassrc-webcam-example.c [hunk 1]
```diff
@@ -54,50 +54,68 @@ init_pipeline ()
...
+    } catch (error) {
+      console.error ("Error accessing webcam:", error);
+    }
+  }
+
+  // Function to draw webcam frames to the OffscreenCanvas
+  function drawToCanvas (stream) {
+    // Create an HTMLVideoElement to play the stream, without adding it to the
+    // DOM
+    const video = document.createElement ('video');
```
Comment (cfoch) [inline gst.wasm/subprojects/samples/webcanvassrc/webcanvassrc-webcam-example.c:96]: I knew this is possible but I wanted to keep the example simple. Other option is ImageCapture (available if setting some flag on Firefox settings), but not sure if this is a good option for video.

Comment (cfoch) [reply gst.wasm/subprojects/samples/webcanvassrc/webcanvassrc-webcam-example.c:96]: Oh, I see the element is not added to the DOM. Nice :-)

File: gst.wasm/subprojects/gst-plugins-web/gst-libs/gst/web/gstwebutils.cpp [hunk 2]
```diff
@@ -541,6 +541,28 @@ gst_web_utils_message_new_propose_object (GstElement *src, gchar *object_name)
...
+
+  (val::global ("HEAPU8").call<val> (
+       "subarray", (guintptr) ret.data, (guintptr) ret.data + ret.len))
+      .call<void> ("set", data);
+
+  return ret;
+}
+
+GstBuffer *
+gst_web_utils_js_array_to_buffer (const val &data)
```
Comment (cfoch) [inline gst.wasm/subprojects/gst-plugins-web/gst-libs/gst/web/gstwebutils.cpp:560]: `gst_web_utils_web_array_to_buffer`? No strong opinion.

File: gst.wasm/subprojects/gst-plugins-web/gst-libs/gst/web/gstwebutils.cpp [hunk 3]
```diff
@@ -541,6 +541,28 @@ gst_web_utils_message_new_propose_object (GstElement *src, gchar *object_name)
...
 
+GByteArray
+gst_web_utils_copy_data_from_js (const val &data)
+{
+  GByteArray ret;
+
+  ret.len = data["length"].as<gsize> ();
+  ret.data = (guint8 *) g_malloc (ret.len);
+
+  (val::global ("HEAPU8").call<val> (
```
Comment (cfoch) [inline gst.wasm/subprojects/gst-plugins-web/gst-libs/gst/web/gstwebutils.cpp:552]: Using EM_ASM would improve readability. Feel free to try it, but performance is more important for sure.

**Review by cfoch — APPROVED** [review]

File: gst.wasm/subprojects/samples/webcanvassrc/webcanvassrc-webcam-example.c (hunk 1 above)
Comment (aslobodeniuk) [reply gst.wasm/subprojects/samples/webcanvassrc/webcanvassrc-webcam-example.c:96]: thanks to chat GPT :)

File: gst.wasm/subprojects/gst-plugins-web/gst-libs/gst/web/gstwebutils.cpp (hunk 2 above)
Comment (aslobodeniuk) [reply gst.wasm/subprojects/gst-plugins-web/gst-libs/gst/web/gstwebutils.cpp:560]: well, we build a buffer from a [TypedArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray), that is a part of JS API. That is why I call it JS array.

We also already have "web" in the gst_web prefix.

**Pull request merged**